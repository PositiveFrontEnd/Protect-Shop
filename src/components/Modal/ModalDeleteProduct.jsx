import React from "react";
import Modal from "./ModalComponents/Modal";
import ModalWrapper from "./ModalComponents/ModalWrapper";
import ModalHeader from "./ModalComponents/ModalHeader";
import ModalClose from "./ModalComponents/ModalClose";
import ModalBody from "./ModalComponents/ModalBody";
import ModalFooter from "./ModalComponents/ModalFooter";
import PropTypes from 'prop-types'

const ModalDeleteProduct = ({ isOpen, onclick, firstClick, secondaryClick }) => {
    return (
        <>
            <ModalWrapper isOpen={isOpen}>
                <Modal>
                    <ModalHeader>
                        <div className="header__title">
                            <p className="title__text">Worning!</p>
                        </div>
                        <ModalClose onclick={onclick} />
                    </ModalHeader>
                    <ModalBody>
                        <p className="modal__body-title title__logOut">Are you sure you want to delete product?</p>
                        <ModalFooter
                            className="footer__email"
                            secondaryText="Yes"
                            firstText="No"
                            firstClassName="modal__button"
                            secondaryClassName="modal__button"
                            firstClick={firstClick}
                            secondaryClick={secondaryClick}
                        />
                    </ModalBody>
                </Modal>
            </ModalWrapper>
        </>
    )
}

ModalDeleteProduct.defaultProps = {
    onclick: () => { },
    isOpen: () => { }
}

ModalDeleteProduct.propTypes = {
    onclick: PropTypes.func,
    isOpen: PropTypes.func,
}

export default ModalDeleteProduct