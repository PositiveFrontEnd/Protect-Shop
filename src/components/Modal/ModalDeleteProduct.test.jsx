import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import ModalDeleteProduct from "./ModalDeleteProduct";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("Тестування компонента  ModalDeleteProduct", () => {
    test("Знімок", () => {
        const mockStore = configureStore([thunk]);
        const initialState = {

        };
        const data = {
            price: 2134,
            onclick: () => { },
            isOpen: () => { },
            secondaryClick: () => { },
            firstClick: () => { },
        };
        const store = mockStore(initialState);
        const modalDeleteProduct = render(
            <Provider store={store}>
                <BrowserRouter>
                    <ModalDeleteProduct data={data} />
                </BrowserRouter>
            </Provider>
        );

        expect(modalDeleteProduct).toMatchSnapshot();
    });
});
