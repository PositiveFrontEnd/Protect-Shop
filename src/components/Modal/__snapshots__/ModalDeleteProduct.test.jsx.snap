// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестування компонента  ModalDeleteProduct > Знімок 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal__wrapper"
      >
        <div
          class="modal"
        >
          <div
            class="modal__header"
          >
            <div
              class="header__title"
            >
              <p
                class="title__text"
              >
                Worning!
              </p>
            </div>
            <button
              class="header__button"
              type="button"
            >
              <svg
                fill="black"
                viewBox="0 0 16 16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                  fill="white"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
          <div
            class="modal__container"
          >
            <div
              class="modal__content modal__content-form"
            >
              <p
                class="modal__body-title title__logOut"
              >
                Are you sure you want to delete product?
              </p>
              <div
                class="footer__email"
              >
                <button
                  class="button-white modal__button"
                  type="button"
                >
                  Yes
                </button>
                <button
                  class="button-black modal__button"
                  type="button"
                >
                  No
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal__wrapper"
    >
      <div
        class="modal"
      >
        <div
          class="modal__header"
        >
          <div
            class="header__title"
          >
            <p
              class="title__text"
            >
              Worning!
            </p>
          </div>
          <button
            class="header__button"
            type="button"
          >
            <svg
              fill="black"
              viewBox="0 0 16 16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.12587 8L2 2.87413L2.87413 2L8 7.12587L13.1259 2L14 2.87413L8.87413 8L14 13.1259L13.1259 14L8 8.87413L2.87413 14L2 13.1259L7.12587 8Z"
                fill="white"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
        <div
          class="modal__container"
        >
          <div
            class="modal__content modal__content-form"
          >
            <p
              class="modal__body-title title__logOut"
            >
              Are you sure you want to delete product?
            </p>
            <div
              class="footer__email"
            >
              <button
                class="button-white modal__button"
                type="button"
              >
                Yes
              </button>
              <button
                class="button-black modal__button"
                type="button"
              >
                No
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
